### PR TITLE
feat: build using ubuntu and add code-coverage

### PR DESCRIPTION
**Pull request by alsami**
* Build using Ubuntu and update appveyor.yml file
* Extend .gitignore to ignore coverage-files
* Add basic codecov.yml
* Adjust Autofac.Build.psm1 to collect code-coverage using coverlet.collector and coverlet.msbuild
* Adjust Autofac.Build.psm1 to install dotnet deps on windows & linux
* Update README.md and add codecov badge
* Remove PackageIconUrl and pack the image directly
* Set Version base to 6.x

### CODE REVIEW

File: appveyor.yml [hunk 1]
```diff
@@ -1,34 +1,39 @@
...
-- provider: NuGet
-  server: https://www.myget.org/F/autofac/api/v2/package
-  api_key:
-    secure: rCUEY75fXN0wxtMy6QL4jCrLdaYbxIBzIXWeN+wEu/XDpyqimzreOc5AH5jMd5ah
-  skip_symbols: true
-  symbol_server: https://www.myget.org/F/autofac/symbols/api/v2/package
-  artifact: MyGet
+image: Ubuntu
+
+version: '6.0.0.{build}'
```
Comment (tillig) [inline appveyor.yml:3]: I don't think we're ready to go to 6.0 yet, are we?

File: build/Autofac.Build.psm1 [hunk 2]
```diff
@@ -1,250 +1,264 @@
...
-    {
-      & dotnet restore ("""" + $Project.FullName + """") --no-cache
-      if($LASTEXITCODE -ne 0)
-      {
-        exit 4
-      }
-    }
-  }
-}
+# EXIT CODES
```
Comment (tillig) [inline build/Autofac.Build.psm1:1]: Bring along the `.psd1` change, too, which is a version number update.

File: .gitignore [hunk 3]
```diff
@@ -1,170 +1,175 @@
...
+[Dd]ebug/
+[Rr]elease/
+x64/
+[Bb]in/
+[Oo]bj/
+
+# Visual Studio 2015 cache/options directory
+.dotnet/
+.vs/
+.vscode/
```
Comment (tillig) [inline .gitignore:25]: Not related to code coverage, but can you also remove the `.vscode` line here? I'll try to get VS Code support for building/working in here and we'll have to remove it anyway.

File: appveyor.yml (hunk 1 above)
Comment (alsami) [reply appveyor.yml:3]: We dropped support for `net461` which is breaking change. Wouldn't the version raise accurate by the definition of semantic-versioning?

File: .gitignore (hunk 3 above)
Comment (alsami) [reply .gitignore]: Done

File: build/Autofac.Build.psm1 (hunk 2 above)
Comment (alsami) [reply build/Autofac.Build.psm1:1]: Done

File: appveyor.yml (hunk 1 above)
Comment (tillig) [reply appveyor.yml]: Oh, yeah, I see it now. It happened in a different pull request so I missed it. You're right, 6 it is.

**Pull request merged**